lab9/phoneBook.js

```diff
@@ -0,0 +1,42 @@
+var phoneBook; // Тут ви зберігаєте записи як хочете
+phoneBook = {cards: [], found: []};
+/*
+   Функція додавання запису в телефонну книгу.
+*/
+phoneBook.add = function (name, phone, email) {
+  // Ваша неймовірна магія тут
+  this.cards.push({name, phone, email});
+};
+
+/*
+   Функція пошуку записв в телефонній книзі.
+   Пошук ведеться по всім полям.
+*/
+phoneBook.find = function (query) {
+  // Ваша магія тут
+  document.write('<span>Знайдені контакти</span><br>'); 
+  phoneBook.cards.map(function(card) {if (card.name.search(query) != -1 || card.phone.search(query) != -1 || card.email.search(query) != -1) phoneBook.found.push(card);});
```

Comment:
This is an awful style :)
Avoid one-liners that are hard to read. Also avoid function bodies without braces, if it's not a simple return or something similar.

Should be something like:

```javascript
phoneBook.cards.map(function(card) {
  if (card.name.search(query) != -1 || card.phone.search(query) != -1 || card.email.search(query) != -1) {
    phoneBook.found.push(card);
  }
});
```

Also read how `map()` works:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map

```
Return value:
A new array with each element being the result of the callback function.
```

Since you're not using the array it returns, you might just use `forEach()` here. Or a `reduce()` if you want to be smart:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce?v=b